### ISSUE TITLE
override plaintext json marshal error when i add a label or annotations

### ISSUE THREAD

**Issue by MrSumeng** [karmada-io/karmada#2556]
**What happened**:
I change the test example of the overridemanager_test.go TestGetMatchingOverridePolicies
```go
for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			//if got := m.getOverridersFromOverridePolicies(tt.policies, tt.resource, tt.cluster); !reflect.DeepEqual(got, tt.wantedOverriders) {
			//	t.Errorf("getOverridersFromOverridePolicies() = %v, want %v", got, tt.wantedOverriders)
			//}
			got := m.getOverridersFromOverridePolicies(tt.policies, tt.resource, tt.cluster)
			deploy := generateDeploymentYaml()
			for _, overriders := range got {
				err := applyPolicyOverriders(deploy, overriders.overriders)
				t.Log(err)
			}
		})
	}
```
Besides, I didn't do anything, but it happened that:
```text
=== RUN   TestGetMatchingOverridePolicies
=== RUN   TestGetMatchingOverridePolicies/OverrideRules_test_1
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'o' in literal false (expecting 'a')
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'h' looking for beginning of value
=== RUN   TestGetMatchingOverridePolicies/OverrideRules_test_2
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'o' in literal false (expecting 'a')
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'a' looking for beginning of value
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'h' looking for beginning of value
=== RUN   TestGetMatchingOverridePolicies/OverrideRules_test_3
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'h' looking for beginning of value
=== RUN   TestGetMatchingOverridePolicies/TargetCluster_and_Overriders_test
    overridemanager_test.go:210: json: error calling MarshalJSON for type v1.JSON: invalid character 'h' looking for beginning of value

```
**What you expected to happen**:

I want know what the way to add labels or annotations and how about edit the yaml when i want to add labels or annotations with plaintext.

**How to reproduce it (as minimally and precisely as possible)**:

only need use this code replace  overridemanager_test.go TestGetMatchingOverridePolicies.

**Anything else we need to know?**:

**Environment**:
- Karmada version: source code from master
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:  go version: go1.18.1 darwin/arm64

**Comment by chaunceyjiang** [comment]
> i want to add labels or annotations with plaintext.

Maybe this can help you.
https://github.com/karmada-io/karmada/blob/master/samples/guestbook/guestbooks-overridepolicy.yaml#L18-L20

**Comment by MrSumeng** [comment]
> 

Thanks. I modified the test case again with you suggest, so it can pass now.
```go
       overriders1 := policyv1alpha1.Overriders{
		Plaintext: []policyv1alpha1.PlaintextOverrider{
			{
				Path:     "/metadata/annotations",
				Operator: "add",
				Value:    apiextensionsv1.JSON{Raw: []byte(`{"foo": "bar"}`)},
			},
		},
	}
```

**Comment by RainbowMango** [comment]
Thanks @chaunceyjiang for your answer.
I think we can close it now according to @MrSumeng feedback above.

I'm thinking if there is a need to introduce overriders for `annotation` and `label`? Then people don't need to write the json path, just like what we did, like `ImageOverride`, `CommandOverrider`, `ArgsOverrider`:

https://github.com/karmada-io/karmada/blob/ab889889faa84b5a1c1ad89241862b642b3cfb4c/Makefile#L91

**Comment by RainbowMango** [comment]
/close
By the way, I added an agenda to [the next community meeting](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit?pli=1#heading=h.odpkzh6pzm1j).

**Comment by chaunceyjiang** [comment]
> I'm thinking if there is a need to introduce overriders for annotation and label?

+1

**Comment by MrSumeng** [comment]
> I'm thinking if there is a need to introduce overriders for annotation and label?
+1
I found that, if the resource label or annotation is nil, we can use this way add it:
```go
       overriders1 := policyv1alpha1.Overriders{
		Plaintext: []policyv1alpha1.PlaintextOverrider{
			{
				Path:     "/metadata/annotations",
				Operator: "add",
				Value:    apiextensionsv1.JSON{Raw: []byte(`{"foo": "bar"}`)},
			},
		},
	}
```
However,  if  the resource label or annotation is not nil, we use this way add it, it will be replaced.
```go
	overriders1 := policyv1alpha1.Overriders{
		Plaintext: []policyv1alpha1.PlaintextOverrider{
			{
				Path:     "/metadata/labels",
				Operator: "add",
				Value:    apiextensionsv1.JSON{Raw: []byte(`{"foo": "bar"}`)},
			},
		},
	}

// the deploy labels is not nil
```
before
<img width="941" alt="image" src="https://user-images.githubusercontent.com/24208988/191646278-9b5e69fd-e780-4bcd-9431-6d8026338ee2.png">
after
<img width="935" alt="image" src="https://user-images.githubusercontent.com/24208988/191646404-85d04bb1-dc26-4044-b779-ec79891cd7ff.png">

now. if I want to add a label or annotation for the resource when it label or annotation is not nil, I need use this way:
```go
	overriders1 := policyv1alpha1.Overriders{
		Plaintext: []policyv1alpha1.PlaintextOverrider{
			{
				Path:     "/metadata/labels/foo",
				Operator: "add",
				Value:    apiextensionsv1.JSON{Raw: []byte(`"bar"`)},
			},
		},
	}
```
To sum up， I think use plaintext override labels or annotation for resource is ambiguity.

**Comment by RainbowMango** [comment]
I get it, the operation `add` actually acts like `replace`, right?
When you try to `add`(append) a new annotation, it actually would replace the whole `annotation`, which means the original annotations will be dropped.

**Comment by MrSumeng** [comment]
Yes. your understanding is very correct.
